x-pack/plugins/integration_assistant/server/graphs/log_type_detection/graph.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ActionsClientChatOpenAI,
+  ActionsClientSimpleChatModel,
+} from '@kbn/langchain/server/language_models';
+import type { StateGraphArgs } from '@langchain/langgraph';
+import { END, START, StateGraph } from '@langchain/langgraph';
+import type { LogFormatDetectionState } from '../../types';
+import { EX_ANSWER_LOG_TYPE } from './constants';
+import { handleLogFormatDetection } from './detection';
+import { SamplesFormat } from '../../../common';
+
+const graphState: StateGraphArgs<LogFormatDetectionState>['channels'] = {
+  lastExecutedChain: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '',
+  },
+  rawSamples: {
+    value: (x: string[], y?: string[]) => y ?? x,
+    default: () => [],
+  },
+  exAnswer: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '',
+  },
+  finalized: {
+    value: (x: boolean, y?: boolean) => y ?? x,
+    default: () => false,
+  },
+  samplesFormat: {
+    value: (x: SamplesFormat, y?: SamplesFormat) => y ?? x,
+    default: () => ({ name: 'json' }),
+  },
+  ecsVersion: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '8.11.0',
+  },
+  results: {
+    value: (x: object, y?: object) => y ?? x,
+    default: () => ({}),
+  },
+};
+
+function modelInput(state: LogFormatDetectionState): Partial<LogFormatDetectionState> {
+  return {
+    exAnswer: JSON.stringify(EX_ANSWER_LOG_TYPE, null, 2),
+    finalized: false,
+    lastExecutedChain: 'modelInput',
+  };
+}
+
+function modelOutput(state: LogFormatDetectionState): Partial<LogFormatDetectionState> {
+  return {
+    finalized: true,
+    lastExecutedChain: 'modelOutput',
+    results: {
+      samplesFormat: state.samplesFormat,
+      parsedSamples: state.rawSamples, // TODO: Add parsed samples
```

Comment:
If we are planning to merge this, why would we return this?

Reply:
This will be something handled after atleast one subgraph is created and samples are parsed into JSON strings. But until then , this is not triggered.